api/context.go

```diff
@@ -459,3 +515,39 @@ func Handle404(w http.ResponseWriter, r *http.Request) {
 	l4g.Error("%v: code=404 ip=%v", r.URL.Path, GetIpAddress(r))
 	RenderWebError(err, w, r)
 }
+
+func CreateOAuthSession(token string) (*model.Session, *model.AppError) {
+	token = model.Md5Encrypt(utils.Cfg.ServiceSettings.TokenSalt, token)
+	accessData := GetAccessData(token, true)
+
+	if accessData == nil {
+		return nil, model.NewAppError("CreateOAuthSession", "Could not find access token", "")
+	}
+
+	if accessData.IsExpired() {
+		if err := RevokeAccessToken(token, true); err != nil {
+			l4g.Error("Encountered an error revoking an access token, err=" + err.Message)
+		}
+		return nil, model.NewAppError("CreateOAuthSession", "Access token is expired", "")
+	}
+
+	var user *model.User
+	if result := <-Srv.Store.User().Get(accessData.UserId); result.Err != nil {
+		return nil, model.NewAppError("CreateOAuthSession", "Could not get the user associated with the access token", "")
+	} else {
+		user = result.Data.(*model.User)
+	}
+
+	session := &model.Session{UserId: user.Id, TeamId: user.TeamId, Roles: user.Roles, AccessToken: accessData.Token}
```

Comment:
It might be better to chat.  Why do we need accessToken,  can't we use SessionId or AltSessionId as the accessToken.